### PR TITLE
feat(cli): add --rollback-dest flag for custom snapshot storage

### PR DESCRIPTION
**Pull request by advaithsujith**
## Summary
- Adds `--rollback-dest <PATH>` CLI flag to redirect rollback snapshot storage to a custom directory (e.g., Docker volume mounts or shared storage) instead of the default `~/.nono/rollbacks/`
- Includes a pre-sandbox permission check that verifies the destination is covered by sandbox write capabilities, failing with an actionable error message if not
- Adds integration tests and edge-case coverage for the new flag

## Changed files
- `crates/nono-cli/src/cli.rs` — new `--rollback-dest` arg on `RunArgs`
- `crates/nono-cli/src/main.rs` — pre-sandbox write permission check + plumbing through `ExecutionFlags`
- `crates/nono-cli/src/rollback_session.rs` — `rollback_root_with_override()` helper
- `tests/integration/test_rollback.sh` — integration tests for the happy/sad path
- `tests/integration/test_rollback_dest_edge_cases.sh` — edge-case tests (nested dirs, symlinks, relative paths, etc.)

## Test plan
- [x] `make test` passes
- [x] Integration test: `--rollback-dest` stores session under custom dir
- [x] Integration test: `--rollback-dest` without write permission fails with clear error
- [x] Edge-case tests pass (nested paths, symlinks, relative paths)
- [ ] Luke to review security implications of the pre-sandbox permission check

### CODE REVIEW

File: crates/nono-cli/src/rollback_session.rs [hunk 1]
```diff
@@ -38,9 +38,10 @@ pub fn rollback_root() -> Result<PathBuf> {
 /// falls back to the default `~/.nono/rollbacks/`.
 pub fn rollback_root_with_override(override_path: Option<&PathBuf>) -> Result<PathBuf> {
     if let Some(path) = override_path {
-        return Ok(path.clone());
+        Ok(path.clone())
+    } else {
+        rollback_root()
```
Comment (lukehinds) [inline crates/nono-cli/src/rollback_session.rs]: The original early-return style is common idiomatic Rust for guard clauses, so there's no compelling reason to change it in a bug fix PR. 

I would roll this back and we are good to go - good PR / work!

**Review by lukehinds — APPROVED** [review]
